### PR TITLE
EC parameter for 3ds max Metal OSL shader

### PR DESCRIPTION
**Pull request by usakhelo**
Hi,
I added energy compensation parameter to max metal OSL shader.
I'll make max plugin PR to support this shortly.

### CODE REVIEW

File: src/appleseed.shaders/src/max/as_max_metal_material.osl [hunk 1]
```diff
@@ -66,6 +66,12 @@ shader as_max_metal_material
     [[
         float min = 1.0
     ]],
+    float                EnergyCompensation = 0.0
```
Comment (dictoon) [inline src/appleseed.shaders/src/max/as_max_metal_material.osl]: Default value should be 1, I believe. Can you check what's the default value in other shaders supporting EC?

Comment (usakhelo) [reply src/appleseed.shaders/src/max/as_max_metal_material.osl]: I checked the studio and if you add MetalBSDF EC parameter is 0

Comment (usakhelo) [reply src/appleseed.shaders/src/max/as_max_metal_material.osl]: in as_metal.osl it's 1.0 by default. I can make max_metal 1 by default too. But what should I do with max materials? Should they be 1 or 0?

Comment (luisbarrancos) [reply src/appleseed.shaders/src/max/as_max_metal_material.osl]: i.e
```
    out_outColor += as_metal(
        distribution,
        Nn,
        tangent,
        in_face_reflectance,
        in_edge_reflectance,
        in_roughness,
        0.0, // specular spread, used in Student's t-MDF only (unused)
        in_anisotropy_amount,
        "energy_compensation", 1.0);   
```
Also, why not use https://github.com/appleseedhq/appleseed/blob/master/src/appleseed.shaders/src/appleseed/as_metal.osl instead?

Comment (luisbarrancos) [reply src/appleseed.shaders/src/max/as_max_metal_material.osl]: @usakhelo ok. About the EC, i really regret not having removed that once i did the PR. If you think for instance what happens with Glass without the EC tables, the amount of energy loss is really dramatic.

**Pull request merged**